dask/base.py

```diff
@@ -1605,4 +1608,51 @@ def clone_key(key, seed):
     raise TypeError(f"Expected str or tuple[str, Hashable, ...]; got {key}")
 
 
+def print_on_compute(msg: str, val: T) -> T:
+    """When the value is computed, print it using distributed.print.
+
+    Note: this may affect performance, since it will change the graph being
+    computed.
+
+    Parameters
+    ----------
+    msg : str
+          Format string for printing out value. Default "%s".
+          It's also possible to use placeholders compatible with str.format(),
+          such as "{}" or "{0}", or the named placeholder "{result}"
+    val : T
+          Lazy value, including a Dask collection, Scalar, etc.
+
+    Returns
+    -------
+    result : T
+          Original value wrapped with an additional Delayed layer
+    """
```

Comment:
- Please add Examples section
- Please include in Sphinx documentation (a link to the API section should be sufficient)